### PR TITLE
Async await script added

### PR DESCRIPTION
**Pull request by avishekjana**
#47

### CODE REVIEW

File: making-api-requests/fetching-data/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Text
+In this lesson we will learn to fetch data in React JS with using `Async Await`. Previously we've made our first API call from a React component, but there we've used `fetch` which returns a Promise. There we've used `.then()` to handle the response and extract the data from it. However, this can lead to "**callback hell**" and make the code difficult to read and maintain.
+
+So let's rewrite the same component using `Async Await`.
+
+# Script
+Using `async` and `await` is a cleaner and more readable way to handle Promises. The `async` keyword is used to declare that a function returns a Promise, and the await keyword is used to wait for the Promise to resolve before continuing with the execution of the code.
+
```
Comment (reenas) [inline making-api-requests/fetching-data/README.md:8]: @avishekjana We can link the WD201 lesson where Async-await was introduced here as well.

File: making-api-requests/fetching-data/README.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+So to start with the implementation, first, we have to clean up all code present in the `ReactPlayground` component.
```
Comment (reenas) [inline making-api-requests/fetching-data/README.md:9]:
```suggestion
So to start with the implementation, first, we have to clean up all code present in the `ReactPlayground` component in our **ReactPlayground.tsx** file.
```

File: making-api-requests/fetching-data/README.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  );
+}
+export default ReactPlayground;
+```
+
+In this component, we've declared the `fetchData` function as `async`, which means it returns a **Promise**. We then use `await` to wait for the Promise returned by `response.json()` to resolve and return the data as a JSON object. 
+
+Then we've used the `setData` function to update the `data`. And once the data updates, it rerenders the component.
+
+So, that was a quick re-fractoring on fetching data in React more efficiently. We've used `async and await` to make the code more readable and easier to follow, as it avoids nested callbacks and allows the code to be written in a more sequential manner.
```
Comment (reenas) [inline making-api-requests/fetching-data/README.md:50]: @avishekjana Can we also mention what is the visible output after writing this code?

Comment (avishekjana) [reply making-api-requests/fetching-data/README.md:54]: Updated a new para to describe the output

File: making-api-requests/fetching-data/README.md (hunk 1 above)
Comment (avishekjana) [reply making-api-requests/fetching-data/README.md:8]: Added link to WD201 lesson.

**Pull request merged**